### PR TITLE
[C-2565] Mark All As Read

### PR DESCRIPTION
**Pull request by RoyalSix**
## Description

https://user-images.githubusercontent.com/16184018/113905747-54dc4400-9788-11eb-84d2-e2f848e1e0c3.mp4

Making a separate issue to fix some more caching/usability issues

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> `Fix [#1]()`

### CODE REVIEW

File: packages/react-inbox/src/components/Message/index.tsx [hunk 1]
```diff
@@ -64,19 +64,20 @@ const Message: React.FunctionComponent<MessageProps> = ({
...
-        label: "View Details",
-        onClick: () => {
-          if (trackingIds?.clickTrackingId) {
-            trackEvent({
-              trackingId: trackingIds?.clickTrackingId,
-            });
-          }
+    () =>
+      [
+        data?.clickAction && {
```
Comment (RoyalSix) [inline packages/react-inbox/src/components/Message/index.tsx:69]: Conditionally rendering this button. If there is no `clickAction` the button does nothing

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
```
Comment (rileylnapier) [inline .husky/pre-commit:1]: cool

File: packages/react-provider/src/hooks/use-track-event.ts [hunk 3]
```diff
@@ -8,8 +8,18 @@ const TRACK_EVENT = `
...
+      ids
+    }
+  }
+`;
+
 const useTrackEvent = () => {
-  return useMutation(TRACK_EVENT);
+  const [, trackEvent] = useMutation(TRACK_EVENT);
+  const [, batchTrackEvent] = useMutation(BATCH_TRACK_EVENT);
+  return [trackEvent, batchTrackEvent];
```
Comment (rileylnapier) [inline packages/react-provider/src/hooks/use-track-event.ts:22]: i feel like we should just return an object here

**Review by rileylnapier — APPROVED** [review]

File: packages/react-provider/src/hooks/use-track-event.ts (hunk 3 above)
Comment (RoyalSix) [reply packages/react-provider/src/hooks/use-track-event.ts:22]: Agree

**Review by rileylnapier — APPROVED** [review]

**Pull request merged**